File: 프로그래머스/C++/주식가격.cpp
#include <string>
#include <vector>
#include <stack>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    
    for(int i = 0; i<prices.size()-1; i++){
        answer.push_back(0);
        for(int j = i+1; j<prices.size(); j++){
            if(prices[i]>prices[j]){
                answer[i] = j-i;
                break;
            }
        }
    }
    
    for(int i = 0; i<prices.size(); i++){
        if(answer[i] == 0)
            answer[i] = prices.size()-i-1;
    }
    answer.push_back(0);
    return answer;
}

File: 프로그래머스/JAVA/주식가격.java
import java.util.Stack;
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int N = prices.length;
        int[] answer = new int[N];
        for(int i = 0; i< N; i++){
            for(int j = i+1; j<N; j++){
                if(prices[i] > prices[j]){
                    answer[i] = j-i;
                    break;
                }
                else
                    answer[i] = j-i;
            }
        }
        return answer;
    }
}
